fairlearn/postprocessing/_tradeoff_curve_utilities.py

```diff
@@ -139,17 +146,16 @@ def _filter_points_to_get_convex_hull(points_sorted):
             else:
                 break
         selected.append(r2)
-    return selected
+    return pd.DataFrame(selected)[[X_KEY, Y_KEY, OPERATION_KEY]]
 
 
 def _interpolate_curve(
     data: pd.DataFrame, x_col: str, y_col: str, content_col: str, x_grid: NDArray
-):
+) -> pd.DataFrame:
     """Interpolates the DataFrame in `data` along the values in `x_grid`.
 
-    Assumes: (1) data[y_col] is convex and non-decreasing in data[x_col]
```

Comment:
It do not necessarily to be non-decreasing (see the [curve on the user-guide ](https://fairlearn.org/v0.11/user_guide/mitigation/postprocessing.html)for instance)

Reply:
That's what I thought too :sweat_smile: